src/fromager/commands/bootstrap.py

```diff
@@ -137,14 +127,13 @@ def write_constraints_file(
 
         # Which parent requirements can use which versions of the dependency we
         # are working on?
-        for dep_version in versions:
-            key = f"{dep_name}=={dep_version}"
-            parent_info = reverse_graph.get(key, [])
-            user_counter += len(parent_info)
-            for parent_version, req in parent_info:
-                for matching_version in req.specifier.filter(versions):
+        for node in nodes:
+            parent_edges = list(node.get_install_edges_to_parent())
```

Comment:
In the original this layer of the loop was getting all of the parents that pointed to the current node so the requirements specifications from those parents could be evaluated against the list of unique versions. So I guess from one perspective it is the edges "to" the parent, but that's because we've reversed them.

I may be bike-shedding on this name. :-)

Reply:
> So I guess from one perspective it is the edges "to" the parent, but that's because we've reversed them.

Yeah thats why I named it like that. Should I change the "to" to "from" i.e. `get_install_edges_from_parent`?

Reply:
It's returning all of the edges into the current node from any parent, right? How about `get_incoming_edges()`?